pkgs/native_assets_cli/lib/src/model/build_output.dart

```diff
@@ -92,12 +99,15 @@ final class BuildOutputImpl implements BuildOutput {
 
   Map<String, Object> toJson(Version version) => {
         _timestampKey: timestamp.toString(),
-        _assetsKey: [
-          for (final asset in _assets) asset.toJson(version),
-        ],
+        _assetsKey: AssetImpl.listToJson(_assets, version),
+        _assetsForLinkingKey:
```

Comment:
Please rev the version to 1.3.0 and document forwards/backwards compatibility in buildoutput_CHANGLOG.md

We can't have compatibility with older Dart SDKs.
What kind of error message do we output when we are invoked with buildConfig.version <= 1.2.0 ? Are we silently ignoring assetsForLink? I believe we should throw an error or exception. UnsupportedError maybe.

Please add a test that exercises this behavior.

Reply:
I think silently ignoring is fine - people won't have an old buildConfig if they use a `link.dart`. right?

Reply:
People can pull a new version of `native_assets_cli` in their project but have an old Dart SDK which does not provide the key (because it has an older version of `native_assets_cli`). Please throw an exception if the `BuildConfig.version <= 1.2.0`.